tests/fgpyo/sam/test_cigar.py

```diff
@@ -31,3 +31,51 @@ def test_bad_index_raises_index_error(index: int) -> None:
 def test_bad_index_raises_type_error(index: int) -> None:
     with pytest.raises(TypeError):
         cigar[index]
+
+
+@pytest.mark.parametrize(
+    ("cigar_string", "start", "end"),
+    {
+        ("10M", 0, 10),
+        ("10M10I", 0, 20),
+        ("10X10I", 0, 20),
+        ("10X10D", 0, 10),
+        ("10=10D", 0, 10),
+        ("10S10M", 10, 20),
+        ("10H10M", 0, 10),
+        ("10H10S10M", 10, 20),
+        ("10H10S10M5S", 10, 20),
+        ("10H10S10M5S10H", 10, 20),
```

Comment:
**issue** the start and end in these three cases should be 20, 30

In each case there are 20 bases of clipping - 10 hard and 10 soft - followed by 10 bases of matched sequence.

Reply:
The bases from hard clips are not in the sequence anymore...that's what make the clip _hard_...so 10,20 is correct.

Reply:
<img width="405" alt="image" src="https://github.com/user-attachments/assets/9b8107dc-4472-45eb-a6b3-464406ac5ec5">

Reply:
This is perhaps a signal that the docstring needs to explicltly explain this - i.e. the offsets are into the read sequence _present on the record_ and as such hard-clips are largely ignored.

Reply:
🤦 

I was thinking in the context of the conditional block above, where the offset is advanced if the operator `is_clipping`

I don't know if the docstring needs to be updated (though wouldn't hurt), but maybe add a note in the respective branch that hard-clipping has zero length on query?